src/java/main/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -1124,6 +1124,7 @@ public void processPacket(ServerCnxn cnxn, ByteBuffer incomingBuffer) throws IOE
             }
             return;
         } else {
+            cnxn.incrOutstandingRequests(h);
```

Comment:
I have 2 observations here which probably don't make a big difference but might worse to consider.
- First, the return statements in the if branches are not required anymore, because there's no more statement at the end of the method anymore,
- Second, moving `cnxn.incrOutstandingRequests(h)` here means that from now on you'll trigger throttling for `sasl` requests too, which was not the case previously. Same for `auth` packets which I believe was done intentionally.

Reply:
hmm right. That return was not there in 3.5.3 where I have spotted the issue. I have missed it when I have moved my change to master.
I see ZOOKEEPER-2785 introduced it. I will update my pr and move incr to the else branch to avoid sasl throttling.